### PR TITLE
Add redirects/rewrite self-serve config

### PR DESCRIPTION
**Pull request by rezrah**
Adds an easier way for maintainers to manage redirects on primer.style. 

Now similar to how Netlify, Vercel and others handle redirects.

### Before

Update `web.config` (XML), which was prone to error and required IIS server domain knowledge

### After: 

Update `redirects.json` with new `name`, `match` (regex) and `destination` fields. GitHub Actions handles the rest ✨ 

#### Complete flow: 

Image showing a flow diagram, asking the reader whether they have previously tested in a staging environment. If not they should merge changes to the staging branch first, before merging to main

### What's changed?

- moved all redirects from web.config into redirects.json
- added a scheme validator for JSON source file
- added a build script to convert JSON to web.config
- add CI/CD step to generate the web.config deterministically and ephemerally on the server (deployed to server, but not committed back to repo)

### Testing

The branch has been deployed to our staging environment.

- 🔗 https://primerstyle-staging.azurewebsites.net/

Here are a sample of existing redirects and rewrites that should continue to work as expected. You can refer to the redirects.json for the full list.

- https://primerstyle-staging.azurewebsites.net/octicons
- https://primerstyle-staging.azurewebsites.net/react
- https://primerstyle-staging.azurewebsites.net/css
- https://primerstyle-staging.azurewebsites.net/brand
- https://primerstyle-staging.azurewebsites.net/octicons/guidelines/design
- https://primerstyle-staging.azurewebsites.net/design/accessibility

### CODE REVIEW

**Review by colebemis — APPROVED** [review]
Looks great! Thanks for doing this, @rezrah 💖

**Comment by rezrah** [comment]
> I was trying to test https://primerstyle-staging.azurewebsites.net/view-components and was getting redirected to react

@jonrohan - thanks for flagging. there was an old redirect for `/components` that wasn't using a more specific regex pattern. As it's a fairly old redirect for primer.style/react, I think it's safe to get rid of it. What do you think?

You can retry the view-component links again now, and should all work as before

https://primerstyle-staging.azurewebsites.net/view-components

(remember to clear your browser cache)

File: redirects.json [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    {
+      "name": "Doctocat",
+      "match": "^doctocat/(.*)",
+      "destination": "https://primer.github.io/doctocat/{R:1}"
+    },
+    {
+      "name": "CLI proxy",
+      "match": "^cli/(.*)",
+      "destination": "https://primer.github.io/cli/{R:1}"
+    },
```
Comment (rezrah) [inline redirects.json:89]: @colebemis given the recent content migrations, can I drop this rewrite now? or do we still need primer.style/cli to be active?

File: redirects.json [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    {
+      "name": "Desktop proxy",
+      "match": "^desktop/(.*)",
+      "destination": "https://primer.github.io/desktop/{R:1}"
+    },
+    {
+      "name": "Contribute proxy",
+      "match": "^contribute/(.*)",
+      "destination": "https://primer.github.io/contribute/{R:1}"
+    },
```
Comment (rezrah) [inline redirects.json:114]: @colebemis similar to preview question, can we drop primer.style/desktop and primer.style/contribute as they now live in primer.style/design?

**Review by jonrohan — APPROVED** [review]
Works great

**Pull request merged**